docs/consume-packages/consuming-packages-authenticated-feeds.md

```diff
@@ -16,11 +16,40 @@ For HTTP feeds, NuGet will make an unauthenticated request, and if the server re
+Moreover, it generally reduces the number of places you need to update when a credential expires or changes.
+If the credential provider supports single sign-on, it may decrease the frequency of logins or the number of places where credentials need to be saved. Refer to the [credential providers](#credential-providers) section for more information.
+
+1. **Encrypted Credentials in nuget.config**: If a credential provider is not available, you should consider using encrypted credentials.
+This method provides an extra layer of security by storing the credentials in an encrypted format.
+For more information, refer to the section on [credentials in *nuget.config* files](#credentials-in-nugetconfig-files).
+
+    > [!NOTE]
+    > :warning: **WARNING** :warning: Be aware that encrypted passwords are only supported on Windows. Moreover, they can only be decrypted on the same machine and by the same user who originally encrypted them.
+
+1. **Using Environment Variable Macros in nuget.config**: If using encrypted credentials is not possible, consider storing the credentials in the *nuget.config* file with environment variable macros.
+This method allows you to reference environment variables that hold the actual credentials.
+For more information, refer to the section on [credentials in *nuget.config* files](#credentials-in-nugetconfig-files).
+
+1. **Using Environment Variables Directly**: As a fallback option, you can store the credentials directly in environment variables.
+However, be aware that this method may offer less visibility and control compared to using environment variable macros in the *nuget.config* file.
+For more information, refer to the section on [credentials in environment variables](#credentials-in-environment-variables).
+
+1. **Clear Text Credentials in NuGet.Config**: It is highly recommended to use one of the previously mentioned options.
+If these options are not feasible, you can store the credentials in the *nuget.config* file.
+However, this option should only be used in environments where no other secure option is available.
+For more information, refer to the section on [credentials in *nuget.config* files](#credentials-in-nugetconfig-files).
+
+    > [!NOTE]
+    > :warning: **WARNING** :warning: Storing credentials in clear text in the *nuget.config* file, especially when saving the file in source control, is risky as it increases the chances of accidental credential leaks.
+    > If you must store credentials in the *nuget.config* file, consider using one of the more secure options mentioned above.
+
+By adhering to these best practices, you can securely authenticate private feeds while minimizing the risk of sensitive information exposure.
 
 The credentials you need to use are determined by the package source.
 Therefore, unless you're using a credential provider, you should check with your package source for what credentials to use.

```

Comment:
This paragraph (package sources define what the correct password/PAT is) doesn't make sense under the security recommendations section. It should be moved to the introduction, as it was before you introduced the new heading.

Reply:
Good catch, Andy. I addressed this comment in the latest commit by moving this paragraph before the new heading.